sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2164,10 +2164,19 @@ class Analyzer(
     }
 
     private def hasNestedGenerator(expr: NamedExpression): Boolean = {
+      def hasUnsupportedInnerGenerator(g: Generator): Boolean = g match {
+        case go: GeneratorOuter => // The only case that can have an inner generator
+          hasUnsupportedInnerGenerator(go.child)
+        case _ =>
+          g.children.exists { _.find {
+            case _: Generator => true
+            case _ => false
+          }.isDefined }
+      }
       CleanupAliases.trimNonTopLevelAliases(expr) match {
-        case UnresolvedAlias(_: Generator, _) => false
```

Comment:
so the method is named `hasNestedGenerator` but never check nested generator?

Reply:
haha, yes. Currently, this method only checks exprs like `add(explode(v), literal(1))`, but does not check exprs like `expolode(explode(v))` handled in this pr.